book/image-controller.md

```diff
@@ -1,2 +1,36 @@
 # Image Controller
 
+# Overview
+Image controller sets up and manages container image repository for an application's component. With this ability, AppStudio provides an isolated environment for image builds of a component that is each component has its own image repository and secret used to push every built image to that repository.
+
+Image controller takes these actions by watching on either specific annotation's change or deletion event of a [Component CR](https://redhat-appstudio.github.io/book/ref/application-environment-api.html#component)
+
+- Setup image repository. Image controller creates an image repository for the Component CR in a remote image registry as well as a robot account which is specific to that repository for image push. A Kubernetes Secret object is also created with that robot account token in order to make it available for build PipelineRun.
+
+- Visibility switch. Image controller is able to switch image repository's visibility between public and private.
+
+- Cleanup. When a Component CR is requested to be deleted, image controller will remove component's image repository and roboto account from the remote registry. The Kubernetes Secret will be removed alongwith the Component CR eventually due to the ownership established between them.
+
+# Dependencies
+Image controller does not depend on other AppStudio services, but a remote image registry. AppStudio services are able to use the resources prepared by image controller, e.g. Build Service makes the Secret available to every build PipelineRun of a component for image push.
+
+# Interface
+Image controller uses annotations to interact with external services.
+
+- `image.redhat.com/generate`: An external service is able to request an image repository for an application component by setting this annotation on the corresponding Component CR. For initial request, the value should include field `visibility` to indicate the visibility of the created image repository in the remote registry, and it can be set again subsequently to change the visibility on demand. Note that, this annotation will be removed once requested operation finishes.
+
+  Here is an example that requests a private image repository:
+
+  ```yaml
+  image.redhat.com/generate: '{"visibility": "private"}'
+  ```
+
+- `image.redhat.com/image`: image controller provides information of prepared resources to external services via this annotation, which includes the image repository URL, the visibility of that repository, and a secret name pointing to the created Kubernetes Secret.
+
+  If something went wrong during the requested operation, this annotation will include a field `Message` with a corresponding error message.
```

Comment:
Will it be good to add an example here like `"message":"failed to generete image repository"`?

Reply:
I would not add such example in this document. Instead, image-controller's own technial document like the README is proper to include necessary examples in detail.